cmd/support-callhome-set.go

```diff
@@ -0,0 +1,158 @@
+	OnUsageError: onUsageError,
+	Action:       mainCallhomeSet,
+	Before:       setGlobalsFromContext,
+	Flags:        callhomeFlags,
+	CustomHelpTemplate: `NAME:
+  {{.HelpName}} - {{.Usage}}
+
+USAGE:
+  {{.HelpName}} TARGET option=on|off
+
+OPTIONS:
+  logs - Push minio log entries to SUBNET
+
+FLAGS:
+  {{range .VisibleFlags}}{{.}}
+  {{end}}
+EXAMPLES:
+  1. Enable logs callhome for cluster with alias 'play'
+     {{.Prompt}} {{.HelpName}} play logs=on
+
+  2. Disable logs callhome for cluster with alias 'play'
+     {{.Prompt}} {{.HelpName}} play logs=off
+`,
+}
+
+func supportedOptions() set.StringSet {
+	return set.CreateStringSet("logs")
+}
+
+func validVals() set.StringSet {
+	return set.CreateStringSet("on", "off")
+}
+
+// checkCallhomeSyntax - validate arguments passed by a user
+func checkCallhomeSetSyntax(ctx *cli.Context) {
+	if len(ctx.Args()) < 2 {
+		cli.ShowCommandHelpAndExit(ctx, "set", 1) // last argument is exit code
+	}
+}
+
+func validateArgs(args []string) error {
+	for _, arg := range args {
+		_, _, e := getOptionKV(arg)
+		if e != nil {
+			return e
+		}
+	}
+	return nil
+}
+
+func mainCallhomeSet(ctx *cli.Context) error {
+	checkCallhomeSetSyntax(ctx)
+
+	aliasedURL := ctx.Args().Get(0)
+	args := ctx.Args().Tail()
+	fatalIf(probe.NewError(validateArgs(args)), fmt.Sprintf("Invalid arguments: %s", strings.Join(args, ",")))
+
+	for _, option := range ctx.Args().Tail() {
+		// options have already been validated, so third (error) return value can be ignored
+		k, v, _ := getOptionKV(option)
+		if k == "logs" {
+			if v == "on" {
+				configureSubnetWebhook(aliasedURL, true)
+			} else {
+				configureSubnetWebhook(aliasedURL, false)
+			}
+		}
+	}
+
+	return nil
+}
+
+func configureSubnetWebhook(aliasedURL string, enable bool) {
+	// Get the alias parameter from cli
+	alias, _ := url2Alias(aliasedURL)
+
+	// Create a new MinIO Admin Client
+	client, err := newAdminClient(aliasedURL)
+	fatalIf(err, "Unable to initialize admin connection.")
+
+	apiKey := getSubnetAPIKeyFromConfig(alias)
+	if len(apiKey) == 0 {
+		err := fmt.Errorf("Please register the cluster first by running 'mc admin subnet register %s'", alias)
+		fatalIf(probe.NewError(err), "Cluster not registered.")
+	}
+
+	enableStr := "off"
+	if enable {
+		enableStr = "on"
+	}
+
+	input := fmt.Sprintf("logger_webhook:subnet endpoint=%s auth_token=%s enable=%s",
+		subnetLogWebhookURL(), apiKey, enableStr)
+
+	// Call set config API
+	restart, e := client.SetConfigKV(globalContext, input)
+	fatalIf(probe.NewError(e), "Unable to set '%s' to server", input)
+
+	// Print set config result
+	printMsg(configSetMessage{
+		targetAlias: aliasedURL,
+		restart:     restart,
+	})
+}
+
+func getOptionKV(input string) (string, string, error) {
+	so := supportedOptions()
+	parts := strings.Split(input, "=")
+	opt := parts[0]
+	if !so.Contains(opt) {
+		return "", "", fmt.Errorf("Invalid option %s. Valid options are %s", opt, so.String())
+	}
+
+	val := parts[1]
```

Comment:
This can crash if the user provide only 'logs'

Reply:
Thanks, fixed.